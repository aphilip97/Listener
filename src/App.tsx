import styled from 'styled-components';
import { Listen, ExtensionsToMimeTypesMap } from '../types';

const Greeting = styled.h1`
  color: hsl(0, 0%, 100%);
`;

const Data = styled.pre`
  color: hsl(0, 0%, 100%);
`;

const map: ExtensionsToMimeTypesMap = {
  webm: 'audio/webm',
  m4a: 'audio/x-m4a',
};

interface AppProps {
  listen: Listen;
}

const App: React.FC<AppProps> = ({ listen }) => {
  return (
    <>
      <Greeting>Listener</Greeting>
      <audio controls autoPlay>
        <source src={listen.src} type={map[listen.ext]} />
      </audio>
      <Data>{JSON.stringify(listen, null, 2)}</Data>
    </>
  );
};

export default App;
