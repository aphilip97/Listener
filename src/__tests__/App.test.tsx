import { render } from '@testing-library/react';
import "@testing-library/jest-dom";

import App from '../App';

test('<App /> matches snapshot', () => {

  const component = render(
    <App />
  );

  expect(component.container).toMatchSnapshot();

});
