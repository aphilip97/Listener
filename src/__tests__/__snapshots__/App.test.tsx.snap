// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div>
  <main
    class="sc-bdvvaa gZRjFI"
  >
    <section
      class="sc-gsDJrp gaOqvD"
    >
      <h1
        class="sc-hKwCoD kvRtqJ"
        data-testid="playlist_title"
      >
        Royalty Free Mixtape For Testing.
      </h1>
      <button
        class="sc-eCImvq iWcURH"
        role="button"
      >
        <span
          class="sc-jRQAMF jcSwbZ"
        >
          Summer Tropical Pop
        </span>
      </button>
      <button
        class="sc-eCImvq iWcURH"
        role="button"
      >
        <span
          class="sc-jRQAMF jcSwbZ"
        >
          Smooth Funk
        </span>
      </button>
      <button
        class="sc-eCImvq iWcURH"
        role="button"
      >
        <span
          class="sc-jRQAMF jcSwbZ"
        >
          Sunday
        </span>
      </button>
    </section>
    <section
      class="sc-dkPtyc bElCjO"
    >
      <audio
        autoplay=""
        controls=""
        data-testid="elem_audio"
      >
        <source
          src="file://wsl%24/Ubuntu-20.04/home/sl0wp0k3/dev/Listener/src/__tests__/assets/royalty_free_mixtape.mp3"
          type="audio/mp3"
        />
      </audio>
    </section>
  </main>
</div>
`;
